### PR TITLE
Add cloud provider name in the intentory

### PR DESCRIPTION
**Pull request by mpagot**
On the Terraform side: add the name of the cloud provider to the inventory. On the Ansible side: make the platform detection optional only in case the variable is not configured from the external (aka the inventory) This change allow to save time in many ansible playbooks.

Ticket: https://jira.suse.com/browse/TEAM-9474

# Verification

## HanaSR

 sle-15-SP5-HanaSr-Azure-Byos-x86_64-Build15-SP5_2024-06-13T02:03:18Z-hanasr_azure_test_sbd@64bit 
 - http://openqaworker15.qa.suse.cz/tests/287220
 - http://openqaworker15.qa.suse.cz/tests/287221
 
#### sle-15-SP5-Qesap-Azure-Byos-x86_64-BuildLATEST_AZURE_SLE15_5_BYOS-qesap_azure_ibsmirror_peering_test@64bit 
 -  http://openqaworker15.qa.suse.cz/tests/287130
 -  http://openqaworker15.qa.suse.cz/tests/287222 :green_circle: http://openqaworker15.qa.suse.cz/tests/287222/logfile?filename=deploy-ansible.pre-cluster.log.txt shows:
    -- all the tasks in the `detect-cloud-platform.yaml` skipped. All out of `TASK [Set Azure from var]` and `TASK [Cloud Platform detection result]` that show `azure`

#### sle-15-SP5-Qesap-Aws-Byos-x86_64-BuildLATEST_AWS_SLE15_5_BYOS-qesap_aws_fencing_native_test@64bit 
 -  http://openqaworker15.qa.suse.cz/tests/287122 . Inventory file http://openqaworker15.qa.suse.cz/tests/287122/file/deploy-inventory.yaml has `cloud_platform_name: aws`
 - http://openqaworker15.qa.suse.cz/tests/287223 :green_circle: from http://openqaworker15.qa.suse.cz/tests/287223/logfile?filename=deploy-ansible.pre-cluster.log.txt shows:
     -- shows all the detect task skipped, all out of `TASK [Probe for AWS 1]` (that is needed to calculate `aws_machine_type`) and `TASK [Set AWS  from var]`

#### sle-15-SP5-Qesap-Aws-Byos-x86_64-BuildLATEST_AWS_SLE15_5_BYOS-qesap_aws_r48xlarge_test@64bit
- http://openqaworker15.qa.suse.cz/tests/287224 :green_circle: in this example `"aws_machine_type": true`

 #### sle-15-SP5-Qesap-Aws-Byos-x86_64-BuildLATEST_AWS_SLE15_5_BYOS-qesap_aws_r5bmetal_test@64bit
 -  http://openqaworker15.qa.suse.cz/tests/287226

### GCP

sle-15-SP5-Qesap-Gcp-Byos-x86_64-BuildLATEST_GCE_SLE15_5_BYOS-qesap_gcp_sapconf_test@64bit
 - http://openqaworker15.qa.suse.cz/tests/287227

sle-15-SP5-Qesap-Gcp-Byos-x86_64-BuildLATEST_GCE_SLE15_5_BYOS-qesap_gcp_sapconf_test@64bit 
 - http://openqaworker15.qa.suse.cz/tests/287228

 #### sle-15-SP5-Qesap-Gcp-Payg-x86_64-BuildLATEST_GCE_SLE15_5_PAYG-qesap_gcp_sapconf_test@64bit 
 -  http://openqaworker15.qa.suse.cz/tests/287126
 -  http://openqaworker15.qa.suse.cz/tests/287129

### CODE REVIEW

**Review by lpalovsky — APPROVED** [review]
LGTM

**Review by alvarocarvajald — APPROVED** [review]
LGTM

File: ansible/playbooks/tasks/detect-cloud-platform.yaml [hunk 1]
```diff
@@ -16,7 +16,7 @@
       changed_when: false
       failed_when: false
       register: probe_aws
-
+      when: cloud_platform_name is undefined or cloud_platform_name | length == 0
```
Comment (mpagot) [inline ansible/playbooks/tasks/detect-cloud-platform.yaml:19]: Like to say: Only attempt autodetection calling the `system-manufacturer` variant of `dmidecode`  if `cloud_platform_name` is not defined by the inventory.yaml

File: ansible/playbooks/tasks/detect-cloud-platform.yaml [hunk 2]
```diff
@@ -25,18 +25,27 @@
       changed_when: false
       failed_when: false
       register: probe_aws_1
+      when: cloud_platform_name is undefined or cloud_platform_name | length == 0 or cloud_platform_name == "aws"
```
Comment (mpagot) [inline ansible/playbooks/tasks/detect-cloud-platform.yaml:28]: Like to say:

> Only attempt autodetection calling the `system-manufacturer` variant of `dmidecode`  if `cloud_platform_name` is not defined by the inventory.yaml. But if `cloud_platform_name` is defined in the inventory like AWS, this command has to be called.

Like to say:

> only skip this step if `cloud_platform_name` is defined in the inventory and value is Azure or Gcp.

File: ansible/playbooks/tasks/detect-cloud-platform.yaml [hunk 3]
```diff
@@ -25,18 +25,27 @@
...
 
     - name: Set AWS machine type fact
       ansible.builtin.set_fact:
         aws_machine_type: true
-      when: probe_aws_1.stdout is match(".*amazon")
+      when:
+        - cloud_platform_name is undefined or cloud_platform_name | length == 0 or cloud_platform_name == "aws"
```
Comment (mpagot) [inline ansible/playbooks/tasks/detect-cloud-platform.yaml:34]: this first when condition is useless and confusing. The only important one is:
```
- probe_aws_1.stdout is match(".*amazon")
```

**Pull request merged**